src/dimred/lsi/test.py

```diff
@@ -0,0 +1,210 @@
+import sys
+import pytest
+import subprocess
+import mudata as mu
+import numpy as np
+
+## VIASH START
+meta = {
+    'resources_dir': 'resources_test',
+    'executable': './target/docker/dimred/lsi/lsi',
+    'config': './src/dimred/lsi/config.vsh.yaml'
+}
+## VIASH END
+
+input_path = f"{meta['resources_dir']}/concat_test_data/e18_mouse_brain_fresh_5k_filtered_feature_bc_matrix_subset_unique_obs.h5mu"
+
+
+'''
+Tests: 
+1. general test 
+2. test HVF
+3. test modality
+4. test layer
+5. test overwrite 
+'''
+
+@pytest.fixture
+def atac_mudata(tmp_path):
+    
+    mdata = mu.read_h5mu(input_path)
+    mdata.mod["atac"].layers["counts"] = mdata.mod["atac"].X
+    mdata.mod["atac"].var["highly_variable"] = np.random.choice([True, False], size=mdata.mod["atac"].n_vars)
+    print(mdata)
+
+    mdata.write(tmp_path / "atac_mudata.h5mu")
+
+    return tmp_path / "atac_mudata.h5mu"
+
+# 1.general test
+def test_lsi(tmp_path):
+    output_path = tmp_path / "output_lsi.h5mu"
+    # run component
+    cmd_args = [
+    meta["executable"],
+     "--input", input_path,
+     "--output", str(output_path),
+     "--obsm_output", "X_test",
+     "--num_components", "30"
+    ]
+    subprocess.run(cmd_args, check=True)
+    
+
+    assert output_path.is_file()
+    data = mu.read_h5mu(output_path)
+    assert "X_test" in data.mod['atac'].obsm
+    assert data.mod["atac"].obsm["X_test"].shape == (data.mod["atac"].n_obs, 30)
+    assert "lsi" in data.mod['atac'].uns
+    assert "LSI" in data.mod['atac'].varm
+
+
+
+# 2.test HVF 
+def test_select_highly_variable_column(atac_mudata, tmp_path):
+    output_path = tmp_path / "output_lsi.h5mu"
+
+    # run component
+    cmd_args = [
+    meta["executable"],
+     "--input", str(atac_mudata),
+     "--output", str(output_path),
+     "--var_input", "highly_variable"
+    ]
+    subprocess.run(cmd_args, check=True)
+    
+    assert output_path.is_file()
+    data = mu.read_h5mu(output_path)
+    assert "X_lsi" in data.mod['atac'].obsm
+    assert data.mod["atac"].obsm["X_lsi"].shape == (data.mod["atac"].n_obs, 50)
+    assert "highly_variable" in data.mod["atac"].var.columns
+    assert "lsi" in data.mod['atac'].uns
+    assert "LSI" in data.mod['atac'].varm
+    assert data.mod["atac"].varm["LSI"].shape == (data.mod["atac"].n_vars, 50)
```

Comment:
Because we've had this pattern of repeatedly creating a random output path, running the component and checking the output, we've implemented some pytest fixtures (specific for openpipelines) in a package that is included in `src/base/openpipelinestestutils`. Also, some other functionality was used in other projects and we've added so we created the [viashpy](https://github.com/viash-io/viashpy) package that also includes some other fixtures (like `run_component`) 

Both are installed by adding this line to the config: 

```
__merge__: [ /src/base/requirements/python_test_setup.yaml, .]
```


I've updated the test below:

- use the `run_component` fixture instead of `subprocess.run`. Its a wrapper around `subprocess.run`, but is adds extra functionality like automatically building the component when you are running the test inline (not using viash test) and adding the options for memory and cpu constraints. You can find its description [here](https://github.com/viash-io/viashpy/blob/4507e2cf1335f8ac3f1bae6b5722629b558c323b/viashpy/testing.py#L158-L170) 
- use `random_h5mu_path` to generate the output path

Could you update the other tests as well?
 
```suggestion
def test_select_highly_variable_column(run_component, random_h5mu_path, atac_mudata, tmp_path):
    output_path = random_h5mu_path()

    # run component
    cmd_args = [
       "--input", str(atac_mudata),
       "--output", str(output_path),
       "--var_input", "highly_variable"
    ]
    run_component(cmd_args)
    
    assert output_path.is_file()
    data = mu.read_h5mu(output_path)
    assert "X_lsi" in data.mod['atac'].obsm
    assert data.mod["atac"].obsm["X_lsi"].shape == (data.mod["atac"].n_obs, 50)
    assert "highly_variable" in data.mod["atac"].var.columns
    assert "lsi" in data.mod['atac'].uns
    assert "LSI" in data.mod['atac'].varm
    assert data.mod["atac"].varm["LSI"].shape == (data.mod["atac"].n_vars, 50)
```